core/src/messages/agent.rs

```diff
@@ -120,7 +115,8 @@ impl JetStreamable for DroneLogMessage {
     }
 }
 
-#[derive(Serialize, Deserialize, Debug, Clone)]
+#[derive(Serialize, Deserialize, Debug, Clone, TypedMessage)]
+#[typed_message("cluster.#cluster.backend.#backend_id.stats")]
```

Comment:
`as_str()` on a `ClusterName` will result in `plane.test` rather than `plane_test` (`.subject_name()`)

Maybe we have another trait, `AsSubjectComponent`, and implement `.as_subject_component()` on our ID types? This would nullify the code savings, but it would help us avoid bugs that came up from using the wrong method (as have occurred before).